packages/yarnpkg-core/sources/Project.ts

```diff
@@ -1690,6 +1720,18 @@ export class Project {
     await xfs.writeFilePromise(installStatePath, serializedState as Buffer);
   }
 
+  async restoreStateForInstall() {
```

Comment:
(Prior context: discussion about making this the installer state rather than the package state)

I'd rename this function `restoreInstallerMetaStates` to make it in line with `restoreInstallState` (it's too bad the names are so similar 🤔 given their similarity, I think it would be worth documenting them both in the same PR)